src/rp2_common/pico_cyw43_driver/cyw43_bus_pio_spi.c

```diff
@@ -85,16 +85,10 @@ __force_inline static uint32_t __swap16x2(uint32_t a) {
 }
 #define SWAP32(a) __swap16x2(a)
 
-#ifndef CYW43_SPI_PIO_PREFERRED_PIO
-#define CYW43_SPI_PIO_PREFERRED_PIO 1
-#endif
-static_assert(CYW43_SPI_PIO_PREFERRED_PIO >=0 && CYW43_SPI_PIO_PREFERRED_PIO < NUM_PIOS, "");
-
 typedef struct {
-    pio_hw_t *pio;
-    uint8_t pio_func_sel;
-    int8_t pio_offset;
-    int8_t pio_sm;
+    PIO pio;
+    uint pio_offset;
```

Comment:
these should be uint8_t

Reply:
That would be awkward because pio_claim_free_sm_and_add_program takes uint*